domain/src/main/kotlin/com/xorker/draw/lock/LockRepository.kt

```diff
@@ -0,0 +1,6 @@
+package com.xorker.draw.lock
+
+interface LockRepository {
+    fun lock(key: String)
+    fun unlock(key: String)
```

Comment:
```suggestion
    fun lock(key: String, call: ()-> Unit)
```

p3. lock / unlock 구분하지 말고,

lock 에서 람다를 받아서 해당 액션 끝나면 해제 해주는건 어떨까요?

Reply:
lock 메서드는 redis에 setnx 명령어를 수행하여 특정 key에 맞게 lock을 거는 작업만 수행하게 됩니다. 만약 설정한 lock time이 만료되었을 때만 unlock을 수행한다면 callback으로 unlock 메서드를 받아서 사용해도 되지만, lock time이 만료되기 전에 unlock을 해야할 경우 직접 호출이 필요하다고 생각합니다. 따라서 lock과 unlock 메서드가 분리되어 독립적으로 사용되는게 맞다는 생각이 들어 의견 남깁니다.

Reply:
Kotlin 확장 함수 중 아래 같은 친구가 있는데 같은 패턴을 쓰면 좋겠다는 이야기 였어요
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html

구현체를 간단히 작성해보면
fun lock(key: String, call: ()-> Unit) {
  // Redis Lock 실행
  
  try {
    call()
  } finally {
    // Redis Unlock
  }
}